### PR TITLE
Get leaderboard results for challenge

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: controllers/challenge.go [hunk 1]
```diff
@@ -141,3 +143,97 @@ func CreateChallenge(ctx context.Context, w http.ResponseWriter, r *http.Request
 	json.NewEncoder(w).Encode(challenge.Key(key))
 	return http.StatusOK, nil
 }
+
+func GetResultsByChallenge(ctx context.Context, w http.ResponseWriter, r *http.Request) (status int, err error) {
+	if r.Method != "GET" {
+		return http.StatusMethodNotAllowed, nil
+	}
+
+	// TODO(victorbalan): Also check if it`s a company
```
Comment (flowlo) [inline controllers/challenge.go]: Add that. It's not too difficult.

File: controllers/challenge.go [hunk 2]
```diff
@@ -141,3 +143,97 @@ func CreateChallenge(ctx context.Context, w http.ResponseWriter, r *http.Request
...
+		cr = append(cr, cro)
+	}
+
+	sort.Sort(cr)
+	json.NewEncoder(w).Encode(cr)
+
+	return http.StatusOK, nil
+}
+
+func getTotalTime(taskOverviews []model.TaskResult) (d time.Duration) {
```
Comment (flowlo) [inline controllers/challenge.go]: Useless named result.

File: controllers/challenge.go [hunk 3]
```diff
@@ -141,3 +143,97 @@ func CreateChallenge(ctx context.Context, w http.ResponseWriter, r *http.Request
...
+	for _, t := range taskOverviews {
+		d += t.CodingTime
+	}
+	return
+}
+
+func getTaskResults(ctx context.Context, challenge model.Challenge, result model.KeyedResult) (results []model.TaskResult, err error) {
```
Comment (flowlo) [inline controllers/challenge.go]: Useless named results.

File: model/challenge_results.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	if r[i].TotalTime == 0 {
+		return false
+	}
+	if r[j].TotalTime == 0 {
+		return true
+	}
+	return r[i].TotalTime < r[j].TotalTime
+}
+
+type ChallengeResult struct {
```
Comment (flowlo) [inline model/challenge_results.go]: Not sure whether we should put this here. You are using it only to encode out of a controller, not for persistence like everything else in the model package.

Comment (victorbalan) [inline model/challenge_results.go]: we already do that in a lot of places..do we need a dto package?

**Pull request closed without merging**